spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/SparkRowLevelOperationsTestBase.java

```diff
@@ -57,6 +60,8 @@ public abstract class SparkRowLevelOperationsTestBase extends SparkExtensionsTes
 
   private static final Random RANDOM = ThreadLocalRandom.current();
 
+  protected static SparkSession otherSpark = null;
```

Comment:
This would add an extra connection to HMS from the Spark built-in client. That's why I reduced the client pool size in tests to 1.